telegram/ext/dispatcher.py

```diff
@@ -275,25 +279,22 @@ def run_async(self, func, *args, update=None, error_handling=True, **kwargs):
         Warning:
             * If you're using ``@run_async``/:meth:`run_async` you cannot rely on adding custom
               attributes to :class:`telegram.ext.CallbackContext`. See its docs for more info.
-            * When using :meth:`run_async` within an error handler (see :meth:`add_error_handler`),
-              you *must* also set :attr:`error_handling` to :obj:`False`, as an exception within
-              :attr:`func` would cause an infinite loop otherwise. Eventual exceptions will then
-              only be logged.
+            * :meth:`run_async` must not be used within an error handler, as an exception within
+              :attr:`func` would cause an infinite loop otherwise.
 
         Args:
             func (:obj:`callable`): The function to run in the thread.
             *args (:obj:`tuple`, optional): Arguments to ``func``.
             update (:class:`telegram.Update`, optional): The update associated with the functions
                 call. If passed, it will be available in the error handlers, in case an exception
                 is raised by :attr:`func`.
-            error_handling (:obj:`bool`, optional): Whether exceptions raised by :attr:`func`
```

Comment:
Removing the documentation does not hide the implementation detail from the user.
I suggest doing something like:
```python

# decorator code
def run_async(func):
    def async_func(*args, **kwargs):
        # call the `_` version
        return Dispatcher.get_instance()._run_async(func, *args, update=None, error_handling=False, **kwargs)

class Dispatcher:
    def run_async(self, func, *args, update=None,  **kwargs): 
        self._run_async(func, *args, update, error_handling=True, **kwargs)

    def _run_async(self, func, *args, update=None, error_handling=True, **kwargs): 
       # current run_async() code comes here

```